src/controller/datacontroller.js

```diff
@@ -198,10 +198,28 @@ export default class DataController {
 			this.model.selection.removeAllRanges();
 			this.model.selection.clearAttributes();
 
+			// Parse data to model and extract markers from parsed document fragment.
+			const { modelItem, markersData } = this.parse( data );
```

Comment:
I know I am usually the one who is against magic, but the possibility of having all data in one object is worth trying.